src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -563,18 +577,33 @@ def create_compile_context(self, target, target_workdir):
       ))
 
   def _runtool_hermetic(self, main, tool_name, args, distribution, tgt=None, input_files=tuple(), input_digest=None, output_dir=None):
-    tool_classpath_abs = self.tool_classpath(tool_name)
+    tool_classpath_abs = self._rsc_classpath
     tool_classpath = fast_relpath_collection(tool_classpath_abs)
 
-    # TODO(#6071): Our ExecuteProcessRequest expects a specific string type for arguments,
-    # which py2 doesn't default to. This can be removed when we drop python 2.
-    str_jvm_options = [text_type(opt) for opt in self.get_options().jvm_options]
-    cmd = [
-            distribution.java,
-          ] + str_jvm_options + [
-            '-cp', os.pathsep.join(tool_classpath),
-            main,
-          ] + args
+    jvm_options = self._jvm_options
+
+    if self._rsc_native_image.use_native_image:
```

Comment:
It would be extremely reasonable to factor out some of this into a helper method in `ZincCompile`.